src/test/java/resources/MyResourceTest.java

```diff
@@ -1,69 +1,63 @@
+package resources;
 
-import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
 import static org.mockito.Mockito.when;
+
 import jakarta.ws.rs.core.Response;
-import jakarta.ws.rs.core.UriBuilder;
-import java.net.URI;
+import millom.sandbox.CustomException.DateFormatIncorrectException;
 import millom.sandbox.service.MessageService;
 import millom.sandbox.service.MessageServiceImpl;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
-import org.junit.runner.RunWith;
 import org.mockito.InjectMocks;
 import org.mockito.Mock;
 import org.mockito.MockitoAnnotations;
-import org.mockito.runners.MockitoJUnitRunner;
 
-@RunWith(MockitoJUnitRunner.class)
 class MyResourceTest {
+
   @Mock
   private MessageService messageServiceMock;
   private MessageService msgServiceImpl;
 
   @InjectMocks
-  MyResource myResource = new MyResource();
+  MyResource myResource;
 
   @BeforeEach
-  void setUp() throws Exception{
+  void setUp() {
     MockitoAnnotations.initMocks(this);
     msgServiceImpl = new MessageServiceImpl();
   }
 
   @Test
   @DisplayName("CONVERT EARTH DATE TO CURIOSITY SOL DATE")
-  void testConvertCuriositySol(){
-//    Response response = Response.
-//        status(Response.Status.OK)
-//        .entity(msgServiceImpl.convertCuriositySol("2022-06-15"))
-//        .build();
+  void testConvertCuriositySol() throws DateFormatIncorrectException {
     when(messageServiceMock.convertCuriositySol("2022-06-15")).thenReturn(3504);
     Response response = myResource.convertCuriositySol("2022-06-15");
     assertEquals(response.getStatus(), 200);
-    assertEquals(response.getEntity(), 3504 );
+    assertEquals(response.getEntity(), 3504);
+
   }
 
   @Test
-  void TestConvertCuriositySolIfDateNull() {
+  void TestConvertCuriositySolIfDateNull() throws DateFormatIncorrectException {
     String date = msgServiceImpl.getTodayDate();
     int actual = msgServiceImpl.convertCuriositySol(date);
 
     when(messageServiceMock.getTodayDate()).thenReturn(date);
     when(messageServiceMock.convertCuriositySol(date)).thenReturn(actual);
-
     Response response = myResource.convertCuriositySol(null);
     assertEquals(response.getStatus(), 200);
     assertEquals(response.getEntity(), actual);
   }
+
   @Test
-  void TestConvertCuriositySolThrowsException() {
-    String date ="2022-";
+  void TestConvertCuriositySolThrowsException() throws DateFormatIncorrectException {
+    String date = "2022-";
     when(messageServiceMock.convertCuriositySol(date)).thenThrow(new RuntimeException());
```

Comment:
That's nit-picking, but still: `MessageService#convertCuriositySol` throws `DateFormatIncorrectException`. Now it's alright, because `MyResource#convertCuriositySol` catches any possible exception. 

Using overarching exceptions can make troubleshooting difficult, so it would be great to catch and test precisely what we expect.

Reply:
Now changed overarching exception to specific exception in MyResource.java
`catch (DateFormatIncorrectException e) {`

And the changes in corresponding test method of MyResourceTest.java are
 ```
@Test
  void TestConvertCuriositySolThrowsException() throws DateFormatIncorrectException {
    String date = "2022-";
    when(messageServiceMock.convertCuriositySol(date)).thenThrow(new DateFormatIncorrectException("Date is not in ISO 8601 format."));
    Response response = myResource.convertCuriositySol(date);
    assertThatThrownBy(()->myResource.convertCuriositySol(date)).isInstanceOf(
            DateFormatIncorrectException.class)
        .hasMessageStartingWith("Date is not in ISO 8601 format.");
        }
```